dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java

```diff
@@ -0,0 +1,958 @@
+        for (Contentlet contentlet : contentlets) {
+            invalidateAll(contentlet);
+        }
+    }
+
+    /**
+     * Invalidate contentlet that is live.
+     *
+     * @param contentlet
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    public void invalidateLive(Contentlet contentlet) throws DotDataException, DotSecurityException {
+        invalidateLive(contentlet, null);
+    }
+
+    /**
+     * Invalidate contentlet that is live.
+     * <p>
+     * NOTE: This method is helpful when you would like to save calls finding the identifier,
+     * otherwise you should use the other method {@link #invalidateLive(Contentlet) invalidateLive}
+     * where you only pass the contentlet
+     * </p>
+     *
+     * @param contentlet
+     * @param identifier of the contentlet, if its null then it will do a search using contentlet
+     *        parameter
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    public void invalidateLive(Contentlet contentlet, Identifier identifier) throws DotDataException, DotSecurityException {
+        Identifier localIdentifier = (identifier == null) ? APILocator.getIdentifierAPI()
+            .find(contentlet) : identifier;
+
+
+        invalidate(contentlet, localIdentifier, false);
+
+
+    }
+
+    /**
+     * Invalidate contentlet that is working
+     *
+     * @param contentlet
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    public void invalidateWorking(Contentlet contentlet) throws DotDataException, DotSecurityException {
+        invalidateWorking(contentlet, null);
+    }
+
+    /**
+     * Invalidate contentlet that is working.
+     * <p>
+     * NOTE: This method is helpful when you would like to save calls finding the identifier,
+     * otherwise you should use the other method {@link #invalidateWorking(Contentlet)
+     * invalidateWorking} where you only pass the contentlet
+     * </p>
+     *
+     * @param contentlet
+     * @param identifier of the contentlet, if its null then it will do a search using contentlet
+     *        parameter
+     * @throws DotDataException
+     * @throws DotSecurityException
+     */
+    public void invalidateWorking(Contentlet contentlet, Identifier identifier) throws DotDataException, DotSecurityException {
+        Identifier localIdentifier = (identifier == null) ? APILocator.getIdentifierAPI()
+            .find(contentlet) : identifier;
+
+
+        invalidate(contentlet, localIdentifier, true);
+
+    }
+
+    private void invalidate(final Contentlet content, final Identifier identifier, final boolean EDIT_MODE)
+            throws DotDataException, DotSecurityException {
+
+        removeContentletFile(content, identifier, EDIT_MODE);
+
+        if (content.getContentType()
+            .baseType() == BaseContentType.HTMLPAGE) {
+            new PageLoader().removePageFile(APILocator.getHTMLPageAssetAPI()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Change this instance-reference to a static reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2209)